Searching Algorithms/Jump Search Algorithm/Python/JumpSearch.py

```diff
@@ -0,0 +1,82 @@
+# ===== ===== =====
+#
+# Theory of Programming
+#
+# Jump Search Algorithm
+# http://theoryofprogramming.com/2016/11/10/jump-search-algorithm/
+# GitHub - https://github.com/VamsiSangam/theoryofprogramming
+# Code Contributor - Saurabh Rane (github.com/ranesr)
+#
+# ===== ===== ===== */
+
+
+# Jump Search Code
+def jump_search(alist, val):
+	
+	length = len(alist)
+	
+	# Calculating jump
+	jump = _integer_sqrt(length)
+	
+	left, right = 0, 0
+
+	while left < length and alist[left] <= val:
+		right = min(length - 1, left + jump)
+		
+		if alist[left] <= val and alist[right] >= val:
+			break
+			
+		left += jump;
+		
+	if left >= length or alist[left] > val:
+		return -1
+		
+	right = min(length - 1, right)
+	
+	i = left
+	while i <= right and alist[i] <= val:
```

Comment:
Consider this:

```
alist = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21]

print(jump_search(alist, 16))
```

In this case, without `alist[i] <= val`, it will run for all elements from left to right in while loop, i.e. 13-19. We can stop after finding an element which is more than `val` (17).